.github/actions/artifact-upload/action.yml

```diff
@@ -0,0 +1,39 @@
+name: 'Artifact upload'
+description: 'Uploads artifact to release'
+inputs:
+  name:
+    description: "Name of the artifact to upload"
+    required: true
+  path:
+    description: "Path of the artifact to upload"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ inputs.name }}
+        path: ${{ inputs.path }}
+
+    - name: Build provenance for attestation (release only)
+      if: github.event_name == 'release'
+      uses: actions/attest-build-provenance@v2
+      with:
+        subject-path: ${{ inputs.path }}
```

Comment:
It looks like the file being attested is now the compressed file, and not the binary. Am I misunderstanding? I think we should be attesting the binary.

Reply:
We can attest both if we want, but we should be at least attesting the binary.

Reply:
Good point, didn't think about that, I'll restore attestation of binary